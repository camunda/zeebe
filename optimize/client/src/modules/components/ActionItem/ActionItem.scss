@use '@carbon/colors';
@import 'shared-styles/shared-styles.scss';

.ActionItem {
  border: 1px solid var(--cds-border-subtle);
  border-radius: 3px;
  background-color: colors.$white;
  width: 100%;
  position: relative;
  min-height: 32px;
  padding: 10px;
  padding-top: 5px;

  .header {
    display: flex;
    width: 100%;
    align-items: center;
    margin-left: auto;

    .type {
      color: var(--cds-text-secondary);
      font-size: 0.75em;
      flex-grow: 1;
    }

    .buttons {
      margin-left: auto;
    }
  }
  .content,
  .warning {
    font-size: 0.75em;
  }

  .content {
    max-width: 100%;
  }

  .warning {
    display: flex;
    align-items: baseline;

    svg {
      margin-left: 0.25rem;
      width: 0.75rem;
      height: 0.75rem;
    }
  }

  &:not(:hover) {
    button {
      opacity: 0;

      &.active,
      &.focus-visible {
        opacity: 1;
      }
    }
  }
}
