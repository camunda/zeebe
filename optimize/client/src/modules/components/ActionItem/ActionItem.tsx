/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {ReactNode} from 'react';
import classnames from 'classnames';
import {Button} from '@carbon/react';
import {Close, Edit, Warning} from '@carbon/icons-react';

import {t} from 'translation';

import './ActionItem.scss';

type ActionItemProps = {
  type?: string;
  warning?: string;
  children?: ReactNode;
  className?: string;
  onClick?: () => void;
  onEdit?: () => void;
};

export default function ActionItem({
  onClick,
  type,
  warning,
  onEdit,
  ...props
}: ActionItemProps): JSX.Element {
  return (
    <div className="ActionItem">
      <div className="header">
        {type && <div className="type">{type}</div>}
        <div className="buttons">
          {!warning && onEdit && (
            <Button
              size="sm"
              kind="ghost"
              onClick={onEdit}
              hasIconOnly
              renderIcon={Edit}
              iconDescription={t('common.edit').toString()}
            />
          )}
          <Button
            size="sm"
            kind="ghost"
            onClick={onClick}
            hasIconOnly
            renderIcon={Close}
            iconDescription={t('common.delete').toString()}
          />
        </div>
      </div>
      <div {...props} className={classnames('content', props.className)}>
        {props.children}
      </div>
      {warning && (
        <span className="warning">
          {warning}
          <Warning />
        </span>
      )}
    </div>
  );
}
