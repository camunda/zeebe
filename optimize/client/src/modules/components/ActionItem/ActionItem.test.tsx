/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {shallow} from 'enzyme';

import ActionItem from './ActionItem';

it('should match snapshot', () => {
  const node = shallow(<ActionItem type="ActionItem type" />);

  expect(node).toMatchSnapshot();
});

it('should render child content', () => {
  const node = shallow(<ActionItem>Some child content</ActionItem>);

  expect(node.find('div.content')).toIncludeText('Some child content');
});

it('should call the onClick handler', () => {
  const spy = jest.fn();
  const node = shallow(<ActionItem onClick={spy}>Content</ActionItem>);

  node.find('Button').simulate('click');

  expect(spy).toHaveBeenCalled();
});

it('should call the onClick handler', () => {
  const spy = jest.fn();
  const node = shallow(<ActionItem onEdit={spy}>Content</ActionItem>);

  node.find('Button').at(0).simulate('click');

  expect(spy).toHaveBeenCalled();
});

it('should show an error warning if specified', () => {
  const node = shallow(<ActionItem warning="There is an error" />);

  expect(node.find('.warning')).toExist();
});

it('should prevent editing the action item if warning prop is specified', () => {
  const spy = jest.fn();
  const node = shallow(<ActionItem warning="There is an error" onEdit={spy} />);

  node.find('Button').at(0).simulate('click');
  expect(spy).not.toHaveBeenCalled();
});
