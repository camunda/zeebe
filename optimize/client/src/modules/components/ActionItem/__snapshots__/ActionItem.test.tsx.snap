// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="ActionItem"
>
  <div
    className="header"
  >
    <div
      className="type"
    >
      ActionItem type
    </div>
    <div
      className="buttons"
    >
      <Button
        hasIconOnly={true}
        iconDescription="Delete"
        kind="ghost"
        renderIcon={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "size": [Function],
            },
            "render": [Function],
          }
        }
        size="sm"
      />
    </div>
  </div>
  <div
    className="content"
  />
</div>
`;
